### ISSUE TITLE
[REC] Balance Druid entering eclipse issue

### ISSUE THREAD

**Issue by Smudgiee** [Hekili/hekili#4248]
### Before You Begin

- [X] I confirm that I have downloaded the latest version of the addon.
- [X] I am not playing on a private server.
- [X] I checked for an [existing, open ticket](https://github.com/Hekili/hekili/labels/recommendation) for this issue and was not able to find one.
- [X] I edited the title of this issue (above) so that it describes the issue I am reporting.
- [X] I am reporting an issue with the default priority included with the specialization (imported or edited priorities are not supported).

### Spec

Druid - Balance

### Describe the Issue

When it suggests to go into Lunar eclipse, it says to press wrath, I press wrath then it shows starfire and towards the end of the cast it will quickly swap to wrath again. It should be doing double wrath and not suggesting to use other abilities between the cast of wrath etc when entering an eclipse according to simc breakdown.

So basically it's suggesting wrath to get ready to enter new eclipse but then it swaps to starfire while casting wrath, then back to wrath suggestion towards end of cast. It will also suggest using other abilities whilst casting first wrath which it shouldn't.

### How to Reproduce

follow the rotation in aoe situtation and check when its about to try and enter an eclipse. I only tested aoe so far, so not sure if it also happens going into solar eclipse. 
Using m+ talents as seen in my log.

### Snapshot (Link)

https://pastebin.com/N752nATE

**Comment by Smudgiee** [comment]
Can confirm this is only using Elune's Chosen with Lunars Calling. Without Lunars Calling it works as intended.

**Comment by joshjau** [comment]
Possible fix, replace lines 46, 47:

```
actions.aoe+=/wrath,if=variable.enter_lunar&(!variable.eclipse|variable.eclipse_remains<cast_time)
actions.aoe+=/starfire,if=!variable.enter_lunar&(!variable.eclipse|variable.eclipse_remains<cast_time)
```

With:

```
actions.aoe+=/wrath,if=variable.enter_lunar&(!variable.eclipse|variable.eclipse_remains<cast_time)&(!prev_gcd.1.wrath|prev_gcd.1.wrath&!prev_gcd.2.wrath)
actions.aoe+=/starfire,if=!variable.enter_lunar&(!variable.eclipse|variable.eclipse_remains<cast_time)&(!prev_gcd.1.starfire|prev_gcd.1.starfire&!prev_gcd.2.starfire)
```

These changes:

- Add conditions to check previous casts using prev_gcd to ensure proper spell sequencing
- Remove the redundant conditions that were causing the flip-flopping behavior
- Ensure that when entering Lunar Eclipse with Lunar's Calling, it will properly suggest double Wrath casts without interruption

You must export the string into the .lua file with the new priorities depending on how you address this, but it may be worth trying until Hekili visits this, as I may be wrong or need a better method.

**Comment by Smudgiee** [comment]
Thanks for the try but it seems to still be doing it sadly. Might also be something to do within in the LUA, not entirely sure because it's smooth for everything other than this build. I will try again later, only had a few mins to check but it didn't seem to change anything.

**Comment by Hekili** [comment]
Please add a new ticket if you are experiencing this since 11.0.7.1
